Setup Dockerfile to build bridge w/ imported msgs

GreatAlexander opened the pull request.

GreatAlexander committed:
Setup Dockerfile to build bridge w/ imported msgs

GreatAlexander committed:
Upgrade Dockerfile w/ stages & convenient scripts

hect95 committed:
Updating Docker scripts and dockerfile to align with Docker multi-stage building convention

- Dockerfile now implements base, dev and runtime stage names convention.
- Docerfile now is aligned with hadolint using COPY instead of ADD, using apt-get instead of apt and few other changes
- run.sh wwill run the runtime container with bash if an argument is defined. `build.sh` scritp is removed due to ambiguity

hect95 committed:
Renaming run.sh into runtime.sh to keep consistency with docker multi-stage build names

hect95 reviewed (dismissed):
Dockerfile and scritps look good to me. Aproved as our initial configuration.

hect95 committed:
Adding privileged docker run

hect95 committed:
Adding ROS 1 network env cfg

hect95 committed:
Limiting parallel build for ros1_bridge

- colcon build now use N amount of workes to build the
	bridge, N = (total host cpu - 2)

GreatAlexander committed:
Add pre-commit and clang file, exclude forked code

GreatAlexander committed:
Run pre-commit and fix some issues (WIP)

hect95 commented on dev.sh:
```diff
@@ -0,0 +1,25 @@
+# ----------------------------------------------------------------
+
+# Build docker image up to dev stage
+DOCKER_BUILDKIT=1 docker build \
+    -t av_bridge_noetic_galactic \
+    -f Dockerfile --target dev .
+
+# Run docker image with local code volumes for development
+docker run -it --rm --net host --privileged \
+    -v /dev/shm:/dev/shm \
+    -v ./ecal_to_ros/ros1:/opt/ros1_msgs_ws/src/ecal_to_ros \
+    -v ./ecal_to_ros/ros2:/opt/ros2_msgs_ws/src/ecal_to_ros \
+    -v ./bin:/opt/ros1_bridge_ws/src/ros1_bridge/bin \
+    -v ./cmake:/opt/ros1_bridge_ws/src/ros1_bridge/cmake \
+    -v ./include:/opt/ros1_bridge_ws/src/ros1_bridge/include \
+    -v ./resource:/opt/ros1_bridge_ws/src/ros1_bridge/resource \
+    -v ./ros1_bridge:/opt/ros1_bridge_ws/src/ros1_bridge/ros1_bridge \
+    -v ./src:/opt/ros1_bridge_ws/src/ros1_bridge/src \
+    -v ./test:/opt/ros1_bridge_ws/src/ros1_bridge/test \
+    -v ./CMakeLists.txt:/opt/ros1_bridge_ws/src/ros1_bridge/CMakeLists.txt \
+    -v ./package.xml:/opt/ros1_bridge_ws/src/ros1_bridge/package.xml \
+    av_bridge_noetic_galactic
```
As discussed docker image name should be: `av_ros1_bridge:latest`

GreatAlexander replied:
Done

hect95 commented on runtime.sh:
```diff
@@ -0,0 +1,22 @@
+fi
+
+# Build docker image only up to base stage
+DOCKER_BUILDKIT=1 docker build \
+    -t av_bridge_noetic_galactic \
+    -f Dockerfile --target runtime .
+
+# Run docker image without volumes
+docker run -it --rm --net host --privileged \
+    -v /dev/shm:/dev/shm \
+    av_bridge_noetic_galactic $CMD
```
As discussed docker image name should be: `av_ros1_bridge:latest`

GreatAlexander replied:
Done

hect95 reviewed (changes requested):
Minor changes

GreatAlexander committed:
Update pre-commit to ignore specific useless info

GreatAlexander committed:
Update docker scripts to use `:latest` tag

GreatAlexander committed:
Fix issues in Dockerfile flagged up by pre-commit

GreatAlexander committed:
Remove og workflow and add our own docker deploy

GreatAlexander commented:
Also brought in our own docker.yml workflow, but we won't be making releases with catkin tools since the forked ros1 bridge pkg has its own changelog / version which we don't want to disturb.
We just need to manually create a tag every time we want to make a release.

hect95 reviewed (approved):
Tested on vehicle and good to go!

PS: I hope we do not need to check this repo again :)
